File: stack with C/stack.c
#include <stdio.h>
#define MAX 5

int array[MAX];
int top = -1;

int push(int value){
	if(top == MAX -1){
		printf("stack overflow\n");
	}else{
		top =top +1;
		array[top]=value;
	}
}

int pop(){
	if(top==-1){
		printf("stack underflow\n");
	}else{
		int item = array[top];
		top =top -1;
		printf("Deleted item is %d",item);
	}
}

int peek(){
	int i;
	if(top==-1){
		printf("\nstack is empty\n");
	}else{
			for(i=0;i<=top;i++){
		printf("%d \t",array[i]);
	}
	}

}

void main(){
	int ch,value;
	
	do{
		printf("\nEnter a choice:\n1)Push 2)Pop 3)Peek 0)exist\n");
		scanf("%d",&ch);
		
		switch(ch){
			case 1:
				printf("Enter the value\n");
				scanf("%d",&value);
				push(value);
				break;
			case 2:
				pop();
				break;
				
			case 3:
				peek();
				break;
			default:
				printf("Invalid choice\n");
				
		}
	}while(ch!=0);
	getch();
}

File: stack/com/Stack.java
package com;

public class Stack {
    private int Max;
    private int[] array;
    private int top;

    public Stack(int size) {
        this.Max = size;
        this.array = new int[Max];
        this.top = -1;
    }

    public void push(int data) {
        if (top == Max - 1) {
            System.out.println("Stack overflow");
        } else {
            top++;
            array[top] = data;
        }
    }

    public int pop() {
        if (top == -1) {
            System.out.println("Stack underflow");
            return -1;
        } else {
            top--;
            return array[top];
        }
    }

    public int peek() {
        if (top == -1) {
            System.out.println("stack is empty");
            return -1;
        } else {
            return array[top];
        }
    }
}